src/plugins/visualize/public/application/editor/editor.js

```diff
@@ -680,16 +679,14 @@ function VisualizeAppController($scope, $route, $injector, $timeout, kbnUrlState
               // Manually insert a new url so the back button will open the saved visualization.
               history.replace(appPath);
               setActiveUrl(appPath);
-              const lastAppType = $scope.getOriginatingApp();
-
-              if (lastAppType === 'dashboards') {
-                const savedVisId = firstSave || savedVis.copyOnSave ? savedVis.id : '';
-                dashboard.addEmbeddableToDashboard({
-                  embeddableId: savedVisId,
-                  embeddableType: VISUALIZE_EMBEDDABLE_TYPE,
-                });
+              if (newlyCreated && embeddable) {
+                embeddable
+                  .getStateTransfer()
+                  .navigateToWithEmbeddablePackage($scope.getOriginatingApp(), {
+                    state: { id: savedVis.id, type: VISUALIZE_EMBEDDABLE_TYPE },
```

Comment:
I believe `EmbeddablePackageState` type should be extended to include embeddable input, to support adding visualizations by value. But this can be done at a second pass, I don't think it should be a blocker for this.

Reply:
Yeah, it could be changed to `EmbeddablePackageByValue | EmbeddablePackageByReference`, with only small modifications to dashboard. This could go in your next PR for visualizations by value.